Implement #53 Get popular board list

nickyanggg opened the pull request:
<!-- 原則上不接受沒有 Issue ID 的 PR。 -->
<!-- We don't accept PRs which has no Issue ID. -->

## 📝 相關的 issue / Related Issues
- 主 issue #51 
- 實作 issue #53

## ⛏ 變更內容 / Details of Changes
<!-- 簡要陳列您的修改 -->
<!-- List down your changes concisely -->
- 實作 delivery/http/route_boards.go 中的 getPopularBoardList
- 實作 usecase/board.go 中的 GetPopularBoards
- TODO: 依 BoardRecord 中的 number_of_user 遞減排序

ars0915 committed:
feat: search board articles by author or title or both

ars0915 committed:
modify: search case insensitive

ars0915 committed:
fix: query parameter to lower

Commit:
add search articles condition: recommendCountGe, modify get query param
method

ars0915 committed:
Merge branch 'development' into feat/search-articles

ars0915 committed:
fix: mock func parameter

ars0915 committed:
fix: file name typo

ars0915 committed:
Add delivery test

DeanXu2357 committed:
添加 openapi 支援

ars0915 committed:
Merge branch 'development' into feat/search-articles

ars0915 committed:
modify: mock part of route boards articles test

ars0915 committed:
move http mock to seperate file

ars0915 committed:
Add usecase board article test and move mock to seperate file

ars0915 committed:
fix golint

DeanXu2357 committed:
token_type 固定為 bearer，改用 enum

Co-authored-by: wagaru <wagaru@hotmail.com>

DeanXu2357 committed:
grant_type 目前只會是 password，改用 enum

Co-authored-by: wagaru <wagaru@hotmail.com>

DeanXu2357 committed:
修正 yaml 格式錯誤

DeanXu2357 committed:
Merge branch 'add_swagger' of github.com:DeanXu2357/Ptt-backend into development

DeanXu2357 committed:
修正 yaml 格式錯誤

DeanXu2357 committed:
改用 docker-compose 啟動 swagger 服務

ars0915 committed:
Add recommend_count_le param and recommend_count cond is set flag

ars0915 committed:
Modify url value set at test

ars0915 committed:
Rename article search condition parameters, modify search condition for
readable

ars0915 committed:
Add test case at get board articles

nickyanggg committed:
Implement #53 Get popular board list.

nickyanggg committed:
add GetPopularBoards mock

nickyanggg committed:
fix /v1/popular-boards reponse format

PichuChen reviewed (approved):
LGTM

ars0915 committed:
modify search article condition for readable

ars0915 committed:
fix: nil slice to empty slice for json return [] instead of null

PichuChen committed:
Merge pull request #24 from ars0915/feat/search-articles

Feat/search articles

DeanXu2357 committed:
更新 docker-compose.yml 需求版本＆文件更新

DeanXu2357 committed:
修正錯字

DeanXu2357 committed:
Merge branch 'development' of https://github.com/Ptt-official-app/Ptt-backend into development

DeanXu2357 committed:
添加測試站 domain

TimoChiang committed:
add route classes test

PHChenGit commented:
LGTM

PichuChen commented on internal/delivery/http/route_users_test.go:
```diff
@@ -86,6 +86,10 @@ func (usecase *MockUsecase) GetBoards(ctx context.Context, userID string) []bbs.
 }
 
+func (usecase *MockUsecase) GetPopularBoards(ctx context.Context) []bbs.BoardRecord {
+	panic("Not implemented")
```
這邊能不能不要用 panic, 而是return error?

nickyanggg replied:
好，待會改

PichuChen commented on internal/usecase/usecase.go:
```diff
@@ -24,6 +24,8 @@ type Usecase interface {
 	// GetBoards returns all board records
 	GetBoards(ctx context.Context, userID string) []bbs.BoardRecord
+	// GetPopularBoards returns top 100 popular board records
+	GetPopularBoards(ctx context.Context) []bbs.BoardRecord
```
回傳的部分建議增加 error

nickyanggg replied:
好～

gn01842919 committed:
feat: implement missing fields of user record

Some fields are missing in the current implementation
since go-bbs does not export those methods yet.

Use a mock to fill the fields

nickyanggg committed:
fix: GetPopularBoards return result & error

nickyanggg commented:
已修改完畢，那因為目前 GetPopularBoards 太過簡單，所以目前 error 值都回傳 nil，之後如果排序的部分可以實作或是 usecase.repo.GetBoards 也有回傳 error 值的話再更改。

PichuChen reviewed (approved):
LGTM

gn01842919 committed:
add error as return value for Repository.GetUsers()

TimoChiang committed:
fix naming for better readability and initialisms

PichuChen committed:
Merge pull request #69 from gn01842919/feature/user_infomation

Implement issue #58 [主線] [PTT] 實作個人看板的使用者資訊 (repository)

PichuChen committed:
Merge pull request #57 from TimoChiang/feature/add_route_classes_tesst

add route classes test

PichuChen committed:
Merge pull request #44 from DeanXu2357/add_swagger

添加 openapi 支援

y2468101216 committed:
增加測試說明

PichuChen committed:
Merge pull request #79 from y2468101216/feature/test_info

增加測試說明

nickyanggg committed:
merge and add popular board record mock test
